build.assets/version-ops/cmd/query-latest/main_test.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-github/v41/github"
+)
+
+func TestGetLatest(t *testing.T) {
+	tests := []struct {
+		desc     string
+		spec     string
+		releases []string
+		wantErr  bool
+		latest   string
+	}{
+		{
+			desc: "pass",
+			spec: "v8",
+			releases: []string{
+				"v8.1.9",
+				"v8.1.10",
+				"v8.0.11",
+			},
+			wantErr: false,
+			latest:  "v8.1.10",
+		},
+		{
+			desc: "fail-bad-spec",
+			spec: "v9",
+			releases: []string{
+				"v8.1.9",
+				"v8.1.10",
+				"v8.0.11",
+			},
+			wantErr: true,
+			latest:  "",
+		},
+		{
+			desc: "pass-prerelease",
+			spec: "v8",
+			releases: []string{
+				"v8.1.10-rc.1",
+				"v8.1.10",
+				"v8.1.10-alpha.1",
+			},
+			wantErr: false,
+			latest:  "v8.1.10",
+		},
+		{
+			desc: "pass-major-minor",
+			spec: "v8.1",
+			releases: []string{
+				"v8.1.9",
+				"v8.2.1",
+				"v8.1.10",
+				"v8.0.11",
+			},
+			wantErr: false,
```

Comment:
If you want to use `testify/require`, you could clean this up a bit. Instead of having `wantErr bool`, you could do something like `assert require.ErrorAssertionFunc`, and each test case would either pass `require.Error` or `require.NoError`.

Then in your loop, no need for an if-statement, just `test.assert(t, err)`

If you want to minimize dependencies, what you have here is totally fine, though you're already pulling in testify so it's not even a net new dependency.

Reply:
Thanks -- I'll check out testify in the future.  I haven't worked with it much yet and it is orthogonal to the purpose of this patch, so I'm gonna decline this for now.

Reply:
Ah, I said the same thing above. Fair enough response.

Reply:
Two folks saying the same thing adds some weight. :smile:  Fixed in 6e10e88.